src/Service/RegisteredMethodManager.php

```diff
@@ -60,6 +60,20 @@ public function registerForMember(Member $member, MethodInterface $method, $data
         // Add it to the member
         $member->RegisteredMFAMethods()->add($registeredMethod);
 
+        if (!MethodRegistry::create()->isBackupMethod($method)) {
+            Notification::create()->send(
```

Comment:
I think this is where @ScopeyNZ was suggesting that you can use dependency injection for the Notification class as a singleton

Reply:
converted.

Reply:
Did you forget to push?

Reply:
[No?](https://github.com/silverstripe/silverstripe-mfa/pull/134/commits/20db45104a9b5b1b44aa27619f95f0180700e264#diff-de30d7d32ad168198aa31a778c164447R64)

Reply:
Simply changing the class from a non-singleton to a singleton isn't implementing dependency injection. Are we treating Notification as a stateless service class? If so, the suggested change would've been like this:

```php
private static $dependencies = ['Notification' => '%$' . Notification::class];
protected $notification;
public function setNotification(Notification $notification) { $this->notification = $notification; return $this; }

/// ...

if (!MethodRegi.....) {
     $this->notification->send(...);
}
```

Note that if this is what we're expecting then I'd also suggest renaming the class to `NotificationService` to indicate that that's what it is rather than an individual model.